### ISSUE TITLE
[JENKINS-23364] Rename hudson-dev:run to jenkins-dev:run

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#11834]
<p>To run Jenkins in development mode, you currently type:</p>

<p>    mvn hudson-dev:run</p>

<p>This command should probably be updated to:</p>

<p>    mvn jenkins-dev:run</p>

<p>See discussion here: <a href="https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4" class="external-link" target="_blank" rel="nofollow noopener">https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4</a></p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kevinburke.jpg" title="kevinburke's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kevinburke">kevinburke</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-23364" target="_blank">Rename hudson-dev:run to jenkins-dev:run</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=albers">albers</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2019-11-12T15:06:59+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>To run Jenkins in development mode, you currently type:</p>

<p>    mvn hudson-dev:run</p>

<p>This command should probably be updated to:</p>

<p>    mvn jenkins-dev:run</p>

<p>See discussion here: <a href="https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4" class="external-link" target="_blank" rel="nofollow noopener">https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4</a></p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers's <a href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=203269&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-203269">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>You seem to be using the maven-jenkins-dev-plugin.
The <tt>jenkins-dev:run</tt> goal is provided by the maven-hpi-plugin. In order to use it, you should add org.jenkins-ci.tools as a plugin group to your maven settings.xml file, as described in the <a href="https://wiki.jenkins-ci.org/display/JENKINS/Plugin+tutorial#Plugintutorial-SettingUpEnvironment" class="external-link" target="_blank" rel="nofollow noopener">plugin tutorial</a>.</p>
</pre>
</details>

</details>
<p>You seem to be using the maven-jenkins-dev-plugin.<br/>
The <tt>jenkins-dev:run</tt> goal is provided by the maven-hpi-plugin. In order to use it, you should add org.jenkins-ci.tools as a plugin group to your maven settings.xml file, as described in the <a href="https://wiki.jenkins-ci.org/display/JENKINS/Plugin+tutorial#Plugintutorial-SettingUpEnvironment" class="external-link" target="_blank" rel="nofollow noopener">plugin tutorial</a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers's <a href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=203272&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-203272">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>updated <a href="https://wiki.jenkins-ci.org/display/JENKINS/Building+Jenkins" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Building+Jenkins</a>: use jenkins-dev:run instead of hudson-dev:run</p>
</pre>
</details>

</details>
<p>updated <a href="https://wiki.jenkins-ci.org/display/JENKINS/Building+Jenkins" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Building+Jenkins</a>: use jenkins-dev:run instead of hudson-dev:run</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers's <a href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=203313&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-203313">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I was incorrect about the maven-hpi-plugin. Both <tt>hudson-dev:run</tt> and <tt>jenkins-dev:run</tt> are provided by maven-jenkins-dev-plugin.</p>

<p><tt>hudson-dev:run</tt> works out of the box because maven-jenkins-dev-plugin explicitly declares hudson-dev as the goalPrefix. For <tt>jenkins-dev:run</tt> to work, the plugin group declaration in settings.xml is necessary.</p>

</pre>
</details>

</details>
<p>I was incorrect about the maven-hpi-plugin. Both <tt>hudson-dev:run</tt> and <tt>jenkins-dev:run</tt> are provided by maven-jenkins-dev-plugin.</p>

<p><tt>hudson-dev:run</tt> works out of the box because maven-jenkins-dev-plugin explicitly declares hudson-dev as the goalPrefix. For <tt>jenkins-dev:run</tt> to work, the plugin group declaration in settings.xml is necessary.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=203324&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-203324">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Harald Albers: Can this be resolved then? Seems to be configuration/documentation issue, and you resolved the latter.</p>
</pre>
</details>

</details>
<p>Harald Albers: Can this be resolved then? Seems to be configuration/documentation issue, and you resolved the latter.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers's <a href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=203526&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-203526">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I'd like to make the <tt>jenkins-dev</tt> prefix available without further configuration, as suggested in the issue.</p>

<p>maven-jenkins-dev-plugin explicitely declares <tt>hudson-dev</tt> as  its <tt>goalPrefix</tt>.
My understanding of <a href="http://maven.apache.org/guides/introduction/introduction-to-plugin-prefix-mapping.html" class="external-link" target="_blank" rel="nofollow noopener">http://maven.apache.org/guides/introduction/introduction-to-plugin-prefix-mapping.html</a> is that this setting makes the default prefix <tt>jenkins-dev</tt> (derived from the <tt>artifactId</tt>), unavailable to default plugin resolution.
Thus, removing the <tt>goalPrefix</tt> declaration should make <tt>jenkins-dev</tt> available.</p>

<p>But this is my first contact with maven plugin programming, so any advice is welcome.</p>
</pre>
</details>

</details>
<p>I'd like to make the <tt>jenkins-dev</tt> prefix available without further configuration, as suggested in the issue.</p>

<p>maven-jenkins-dev-plugin explicitely declares <tt>hudson-dev</tt> as  its <tt>goalPrefix</tt>.<br/>
My understanding of <a href="http://maven.apache.org/guides/introduction/introduction-to-plugin-prefix-mapping.html" class="external-link" target="_blank" rel="nofollow noopener">http://maven.apache.org/guides/introduction/introduction-to-plugin-prefix-mapping.html</a> is that this setting makes the default prefix <tt>jenkins-dev</tt> (derived from the <tt>artifactId</tt>), unavailable to default plugin resolution.<br/>
Thus, removing the <tt>goalPrefix</tt> declaration should make <tt>jenkins-dev</tt> available.</p>

<p>But this is my first contact with maven plugin programming, so any advice is welcome.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers's <a href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=203572&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-203572">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><tt>mvn jenkins-dev:run</tt> ist available only after applying special maven configuration while <tt>mvn hudson-dev</tt> works out of the box.</p>

<p>This behavior should be reversed.</p>
</pre>
</details>

</details>
<p><tt>mvn jenkins-dev:run</tt> ist available only after applying special maven configuration while <tt>mvn hudson-dev</tt> works out of the box.</p>

<p>This behavior should be reversed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers's <a href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=204356&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-204356">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>created <a href="https://github.com/jenkinsci/maven-hudson-dev-plugin/pull/1" class="external-link" target="_blank" rel="nofollow noopener">PR #1</a></p>
</pre>
</details>

</details>
<p>created <a href="https://github.com/jenkinsci/maven-hudson-dev-plugin/pull/1" class="external-link" target="_blank" rel="nofollow noopener">PR #1</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev's <a href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=204357&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-204357">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Reversing the behavior could be a breaking change.
Is it possible to a support both call options out-of-the-box (e.g. by parametrizing the prefix)?</p>
</pre>
</details>

</details>
<p>Reversing the behavior could be a breaking change.<br/>
Is it possible to a support both call options out-of-the-box (e.g. by parametrizing the prefix)?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers's <a href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=204367&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-204367">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>The proposed change only covers interactive invocation from the command line (<tt>mvn jenkins-dev:run</tt> vs <tt>mvn hudson-dev:run</tt>). It has no impact on how the goals are called from existing build scripts (configuring goal exectuion in certain phases in pom.xml).</p>

<p>As far as I know, there is no way of parameterising the prefix. There can only be one.</p>

<p>In order to support both prefixes, the plugin has to be renamed back to <em>maven-hudson-dev-plugin</em> and a <em>goalPrefix</em> of <em>jenkins-dev</em> has to be configured in the plugin. But this would impact existing builds.</p>
</pre>
</details>

</details>
<p>The proposed change only covers interactive invocation from the command line (<tt>mvn jenkins-dev:run</tt> vs <tt>mvn hudson-dev:run</tt>). It has no impact on how the goals are called from existing build scripts (configuring goal exectuion in certain phases in pom.xml).</p>

<p>As far as I know, there is no way of parameterising the prefix. There can only be one.</p>

<p>In order to support both prefixes, the plugin has to be renamed back to <em>maven-hudson-dev-plugin</em> and a <em>goalPrefix</em> of <em>jenkins-dev</em> has to be configured in the plugin. But this would impact existing builds.</p>
